lib/providers/nutrition.dart

```diff
@@ -417,17 +462,86 @@ class NutritionPlansProvider with ChangeNotifier {
     final data = await baseProvider.fetchPaginated(
       baseProvider.makeUrl(
         _nutritionDiaryPath,
-        query: {'plan': plan.id.toString(), 'limit': '999', 'ordering': 'datetime'},
+        query: {
+          'plan': plan.id.toString(),
+          'limit': '999',
+          'ordering': 'datetime'
+        },
       ),
     );
 
-    plan.diaryEntries = [];
+    plan.logs = [];
     for (final logData in data) {
       final log = Log.fromJson(logData);
       final ingredient = await fetchIngredient(log.ingredientId);
-      log.ingredient = ingredient;
-      plan.diaryEntries.add(log);
+      log.ingredientObj = ingredient;
+      plan.logs.add(log);
     }
     notifyListeners();
   }
+
+  /// NEW: Notification Manager for meals
+  /// Schedule notifications for all meals in the plans
+  Future<void> _initializeNotifications() async {
+    var androidPlatformChannelSpecifics =
+    AndroidInitializationSettings('@mipmap/ic_launcher');
+    var initializationSettings =
+    InitializationSettings(android: androidPlatformChannelSpecifics);
+    await flutterLocalNotificationsPlugin.initialize(initializationSettings);
+    _scheduleNotifications();
+  }
+
+  Future<void> _scheduleNotifications() async {
+    await flutterLocalNotificationsPlugin.cancelAll();
+    await _scheduleSingleNotification();
+  }
+
+  Future<void> _scheduleSingleNotification() async {
+    var androidPlatformChannelSpecifics = AndroidNotificationDetails(
+      'wger_channel',
+      'wger_channel',
+      'Channel for wger notifications',
+      importance: Importance.max,
+      priority: Priority.high,
+      ticker: 'ticker',
+    );
+
+    var platformChannelSpecifics = NotificationDetails(
+      android: androidPlatformChannelSpecifics,
+    );
+
+    // Convert DateTime to TZDateTime
+    var scheduledDate = tz.TZDateTime.now(tz.local).add(Duration(seconds: 5));
+
+    // Create a mutable PendingIntent for Android S+
+    final androidNotification = AndroidNotificationDetails(
+      'wger_channel',
+      'wger_channel',
+      'Channel for wger notifications',
+      importance: Importance.max,
+      priority: Priority.high,
+      ticker: 'ticker',
+    );
+    final notificationDetails = NotificationDetails(
+        android: androidNotification);
+    final androidPlugin = FlutterLocalNotificationsPlugin();
+    await androidPlugin.zonedSchedule(
+      0,
+      'Nutrition Reminder',
+      'It\'s time for your meal!',
+      scheduledDate,
+      notificationDetails,
+      androidAllowWhileIdle: true,
+      uiLocalNotificationDateInterpretation:
+      UILocalNotificationDateInterpretation.absoluteTime,
+      payload: 'meal', // Use payload to identify the notification type
+    );
+    print('notification scheduled!!!');
+  }
 }
+
+  // Schedule notifications // TO DO
```

Comment:
i suppose we would also need to cancel pending notifications when people delete meals, change their time, switch to a different plan etc